backend/api/Services/BlobService.cs

```
@@ -0,0 +1,74 @@
+﻿using Api.Options;
+using Azure;
+using Azure.Identity;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Extensions.Options;
+namespace Api.Services
+{
+    public interface IBlobService
+    {
+        public Task<byte[]> DownloadBlob(string blobName, string containerName, string accountName);
+
+        public AsyncPageable<BlobItem>? FetchAllBlobs(string containerName, string accountName);
+    }
+
+    public class BlobService : IBlobService
+    {
+        private readonly IOptions<AzureAdOptions> _azureOptions;
+        private readonly ILogger<BlobService> _logger;
+
+        public BlobService(ILogger<BlobService> logger, IOptions<AzureAdOptions> azureOptions)
+        {
+            _logger = logger;
+            _azureOptions = azureOptions;
+        }
+
+        public async Task<byte[]> DownloadBlob(string blobName, string containerName, string accountName)
+        {
+            var blobContainerClient = GetBlobContainerClient(containerName, accountName);
+            var blobClient = blobContainerClient.GetBlobClient(blobName);
+
+            await using var stream = await blobClient.OpenReadAsync();
+
+            byte[] result = new byte[stream.Length];
+
+            // ReSharper disable once MustUseReturnValue
+            await stream.ReadAsync(result);
+
+            return result;
+        }
+
+        public AsyncPageable<BlobItem>? FetchAllBlobs(string containerName, string accountName)
+        {
+            var blobContainerClient = GetBlobContainerClient(containerName, accountName);
+
+            AsyncPageable<BlobItem>? blobs;
+            try
+            {
+                blobs = blobContainerClient.GetBlobsAsync(BlobTraits.Metadata);
+            }
+            catch (RequestFailedException e)
+            {
+                _logger.LogWarning("Failed to fetch blob items because: {ErrorMessage}", e.Message);
+                return null;
```

Return error message instead of null?